app/src/main/java/com/framgia/soundclound/screen/playtrack/PlayTrackViewModel.java

```diff
@@ -32,32 +82,167 @@ public String getArtist() {
+        notifyPropertyChanged(0);
+    }
+
+    public ServiceConnection getServiceConnection() {
+        return mServiceConnection;
+    }
+
+    public boolean isMusicBound() {
+        return mMusicBound;
+    }
 
+    public void startPlayMusic() {
+        mMusicService.play();
+        TimeAsysnTask asyncTask = new TimeAsysnTask();
+        asyncTask.execute();
+    }
+
+    public void onClickPause(View view) {
+        ImageView imageView = (ImageView) view;
+        if (mMusicService.isPlay()) {
+            mMusicService.pause();
+            imageView.setImageResource(android.R.drawable.ic_media_pause);
+        } else {
+            mMusicService.start();
+            imageView.setImageResource(android.R.drawable.ic_media_play);
+        }
+    }
+
+    public void onClickNext(View view) {
+        mMusicService.next();
+        updateUI();
     }
 
     public void onClickShuffle(View view) {
+        mMusicService.setShuffle();
+        ImageView imageView = (ImageView) view;
+        int colorSelect;
+        if (mMusicService.isShuffle()) {
+            colorSelect = R.color.color_orange;
+            imageView.setImageResource(android.R.drawable.ic_media_pause);
+        } else {
+            colorSelect = R.color.color_gray;
+            imageView.setImageResource(android.R.drawable.ic_media_play);
+        }
+        imageView.setColorFilter(colorSelect);
+    }
+
+    public void onClickLoop(View view) {
+        mMusicService.setRepeat();
+        ImageView imageView = (ImageView) view;
+        int colorSelect;
+        if (mMusicService.isRepeat()) {
+            colorSelect = R.color.color_orange;
+        } else {
+            colorSelect = R.color.color_gray;
+        }
+        imageView.setColorFilter(colorSelect);
+    }
 
+    public void onClickPrevious(View view) {
+        mMusicService.prev();
+        updateUI();
     }
 
-    public void onClickLike(View view) {
+    private void updateUI() {
+        List<Track> tracks = SharePreferences.getInstance().getListTrack();
+        mIndexCurrentTrack = SharePreferences.getInstance().getIndex();
+        if (tracks != null) {
+            setTrack(tracks.get(mIndexCurrentTrack));
+        }
+    }
 
+    public void onClickBack(View view) {
+        mOnClickTrackListener.onClickBack(view);
     }
 
-    public void onClickDownload(View view) {
+    @Override
+    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+
+    }
 
+    @Override
+    public void onStartTrackingTouch(SeekBar seekBar) {
+//         asyncTask.cancel(true);
     }
 
+    @Override
+    public void onStopTrackingTouch(SeekBar seekBar) {
+        int currentPosition = StringUtil.progressToTimer(
+                seekBar.getProgress(), mMusicService.getDuration());
+        mMusicService.seekto(currentPosition);
+    }
+
+    /**
+     * AsyncTask update time
+     */
+
+    public class TimeAsysnTask extends AsyncTask<Void, Void, Void> {
```

Comment:
WARNING: [StaticFieldLeak] This AsyncTask class should be static or leaks might occur (com.framgia.soundclound.screen.playtrack.PlayTrackViewModel.TimeAsysnTask)
 A static field will leak contexts.

Non-static inner classes have an implicit reference to their outer class. If that outer class is for example a `Fragment` or `Activity`, then this reference means that the long-running handler/loader/task will hold a reference to the activity which prevents it from getting garbage collected.

Similarly, direct field references to activities and fragments from these longer running instances can cause leaks.

ViewModel classes should never point to Views or non-application Contexts.